### DISCUSSION TITLE
Kernel Panic when booting from USB installation

### DISCUSSION THREAD

**Discussion by GoopyKibbles** [zbm-dev/zfsbootmenu#610] [Questions]
Hi all, 
recently I've been Installing archlinux onto a usb hard drive with native encryption enabled. I have the root system on a zfs pool and   I've placed a zbm recovery prebuilt EFI executable in my EFI System Partition. It worked quite well at first, on a PC, however when trying to boot on a MacBook Pro 11,1 there's always a kernel panic. I searched for this issue and tried to mask the teardown script but it didn't work.

When booting from my usb drive using zfsbootmenu, the commandline shows multiple lines of `No pools available to import`, and then it shows `Importing pool zroot`, and finally it can't import the pool and a kernel panic happens. However this never happened on my other PC.

Any help provided will be greatly appreciated!

(First time to start a discussion on GitHub so please point it out if I did anything wrong!)

**Comment by zdykstra** [comment]
I'm not quite sure what could be happening here. I imagine there's some weirdness with things being on a USB drive and the bus not responding quickly. I imagine there's no possibility of using the internal storage for Arch?

**Reply by GoopyKibbles** [reply]
Currently running macOS on my internal SSD and my main goal is to bring my arch installation around as some sort of alternative OS. Not a severe issue for me though. Recently switched back to grub2 and it seems that everything works just fine. Still not able to figure out what's going on yet, cause according to my drive's indicator light the disk gets *disconnected* (even though that I DID mask the usb teardown script) when zbm starts the new kernel. Have no idea right now, though.
